c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -801,6 +958,58 @@ where
             return Ok(());
         }
 
+        if from.option && from.own != to.own {
+            // Downgrade ownership before unwrapping the `Option` when possible.  This can avoid
+            // moving/consuming the input.  For example, if the `from` type is `Option<Box<T>>` and
+            // `to` is `&mut T`, we start by calling `p.as_mut().as_deref()`, which gives
```

Comment:
Wouldn't that be `p.as_deref_mut()`?

Reply:
Fixed.  I originally implemented rewriting to `p.as_mut().as_deref_mut()`, but this doesn't produce the right output type.  I fixed the rewriting to produce `p.as_deref_mut()`, but forgot to update the comment.